src/app/mork/celery/tasks/edx.py

```diff
@@ -0,0 +1,97 @@
+"""Mork Celery edx tasks."""
+
+from logging import getLogger
+from uuid import UUID
+
+import httpx
+from celery import chain
+from sqlalchemy.exc import SQLAlchemyError
+
+from mork.celery.celery_app import app
+from mork.conf import settings
+from mork.edx.mysql import crud
+from mork.edx.mysql.database import OpenEdxMySQLDB
+from mork.exceptions import UserStatusError
+from mork.models.users import DeletionStatus, ServiceName
+
+logger = getLogger(__name__)
+
+
+@app.task
+def delete_edx_platform_user(user_id: UUID, email: str):
+    """Main task to delete user from the edX service."""
+    workflow = chain(
+        check_user_status.si(user_id=user_id, service=ServiceName.EDX),
+        delete_edx_mysql_user.si(email=email),
+        delete_edx_mongo_user.si(email=email),
+        update_user_status.si(user_id=user_id, service=ServiceName.EDX),
+    )
+    workflow.delay()
+
+
+@app.task(
+    bind=True,
+    retry_kwargs={"max_retries": settings.DELETE_MAX_RETRIES},
+)
+def delete_edx_mysql_user(self, email: str):
+    """Task to delete user's data from edX MySQL database."""
+    logger.debug("Deleting user's data from edX MySQL")
+
+    db = OpenEdxMySQLDB()
+    crud.delete_user(db.session, email=email)
+    try:
+        db.session.commit()
+    except SQLAlchemyError as exc:
+        db.session.rollback()
+        logger.error("Failed to delete user with email %s", email)
+        raise self.retry(exc=exc) from exc
+    finally:
+        db.session.close()
+
+
+@app.task(
+    bind=True,
+    retry_kwargs={"max_retries": settings.DELETE_MAX_RETRIES},
+)
+def delete_edx_mongo_user(email: str):  # noqa: ARG001
+    """Task to delete user's data from edX MongoDB database."""
+    logger.debug("Deleting user's data from edX MongoDB")
+    pass
+
+
+@app.task
+def check_user_status(user_id: UUID, service: ServiceName):
+    """Check user is to be deleted on the service."""
+    logger.debug("Checking that user status is TO_DELETE")
+
+    url = f"{settings.SERVER_URL}/v1/users/{user_id}/status/{str(service)}"
+    auth_headers = {"X-API-Key": f"{settings.API_KEYS[0]}"}
+    response = httpx.get(url, headers=auth_headers)
```

Comment:
You're right, it will probably be used when removing Brevo entries too.
Created functions in a `celery/utils.py` file as I didn't find a great name for it